clang/lib/Driver/ToolChains/CommonArgs.cpp

```diff
@@ -2625,6 +2625,11 @@ void tools::AddStaticDeviceLibs(Compilation *C, const Tool *T,
                                 llvm::opt::ArgStringList &CC1Args,
                                 StringRef Arch, StringRef Target,
                                 bool isBitCodeSDL) {
+  
+  // Check if offload libraries are disabled
+  if (!DriverArgs.hasFlag(options::OPT_offloadlib, options::OPT_no_offloadlib,
+                          true))
+    return;
```

Comment:
The `-[-no]-offloadlib` option is not meant for disabling SDL and I think there is an option to disable SDL. The option is meant for enabling/disabling linking of runtime device library which in case of the chipStar is the `hipspv-spirv64.bc` (which is handled [here](https://github.com/llvm/llvm-project/blob/74687180dde07312521db09c6f6454fe9d1e5662/clang/lib/Driver/ToolChains/HIPSPV.cpp#L203-L208)).